### DISCUSSION TITLE
Black foreground, white background

### DISCUSSION THREAD

**Discussion by onurdumangoz** [dlemstra/Magick.NET#1630] [Help] [answered]
I have image with transparent background. I want convert it to, white background and solid black foreground.

For examle:
I need to convert an image with a transparent background to an image with a white background and black foreground.
![im-logo-1](https://github.com/dlemstra/Magick.NET/assets/40442783/be681be8-f39b-40f6-b0f3-e39e80a8a911)
![im-logo-2](https://github.com/dlemstra/Magick.NET/assets/40442783/e5ac7102-d79e-44f7-b31d-8eee2cd3c6da)

I can do this in CLI with this code:
`magick im-logo-1.png xc:black -channel RGB -clut -alpha remove im-logo-2.png`

How can I do this in Magick.NET?

**Comment by dlemstra** [comment, marked as answer]
It looks like you found a bug in the code. While writing an answer I figured out there was a bug in the code of the `.Clut` method. This has been fixed an will become available in the next release. After that release you can do this:

```C#
using var image = new MagickImage("im-logo-1.png");
using var clutImage = new MagickImage("xc:black", 1, 1);
image.Clut(clutImage, Channels.RGB); // This overload will also be added and in the current release the channels are used incorrectly.
image.Alpha(AlphaOption.Remove);
image.Write("im-logo-2.png");
```

**Reply by onurdumangoz** [reply]
Thank you for your work. I'm waiting for the update.

**Reply by alexlezam** [reply]
According to the answer in the update, Channels.RGB is going to be added, so that it only does it only where it is not transparent?

**Reply by dlemstra** [reply]
That is correct. It means this operation will only change the Red Green and Blue channels.
